swing/src/net/sf/openrocket/gui/scalefigure/RocketFigure.java

```diff
@@ -315,6 +315,12 @@ public void paintComponent(Graphics g) {
 		
 	}
 	
+	@Override
+	public void updateFigure() {
+		updateShapes(this.figureShapes);
```

Comment:
if you're updating the shapes on every change, does that make the `updadeShapes` call at [194 ](https://github.com/openrocket/openrocket/blob/ba37faa16eda9344190177d60c51b56c74474a75/swing/src/net/sf/openrocket/gui/scalefigure/RocketFigure.java#L194)  redundant?

Reply:
Good question, I'll have to look at it a bit more and get back to you. I originally added this in as the figure was lacking any of the figureShapes until the paintComponent. I left the call on line 194 b/c the updateShapes method will also handle any transform changes which are applied as part of the paintComponent method. However, I'll look at this code path a bit more and come back with more information.

Reply:
@teyrana after further review and testing, overriding the updateFigure method is not necessary here. I'll push an additional commit to address this.